src/bin/pgcopydb/follow.c

```diff
@@ -388,11 +394,19 @@ follow_prepare_mode_switch(StreamSpecs *streamSpecs)
 		}
 	}
 
-	/* first empty the transform queue, where a STOP message has been sent */
-	if (!stream_transform_from_queue(streamSpecs))
+	/*
+	 * If the current mode is replay and the previous mode was catch-up, then
+	 * before proceeding, we need to empty the transform queue where the STOP
+	 * message was sent.
+	 */
+	if (currentMode == STREAM_MODE_REPLAY &&
+		previousMode == STREAM_MODE_CATCHUP)
```

Comment:
I believe the condition  that previousMode == CATCHUP is all we need to make that decision, whatever the current/next mode is going to be, right?

Reply:
Yep, I also had a similar thought, but then by "future-proofing" I assumed we only want to empty the queue if both the conditions are met (which doesn't make much sense but who knows what the future holds :). Let me change this right away.